packages/core/content-manager/server/controllers/relations.js

```diff
@@ -1,64 +1,96 @@
 'use strict';
 
-const { prop, pick } = require('lodash/fp');
+const { prop, isEmpty, isNil } = require('lodash/fp');
+const { convertFiltersQueryParams } = require('@strapi/utils/lib/convert-query-params');
+const { hasDraftAndPublish } = require('@strapi/utils').contentTypes;
 const { PUBLISHED_AT_ATTRIBUTE } = require('@strapi/utils').contentTypes.constants;
 
 const { getService } = require('../utils');
+const { validateFindNew } = require('./validation/relations');
 
 module.exports = {
-  async find(ctx) {
+  async findNew(ctx) {
     const { model, targetField } = ctx.params;
-    const { _component, ...query } = ctx.request.query;
-    const { idsToOmit } = ctx.request.body;
 
-    if (!targetField) {
-      return ctx.badRequest();
-    }
+    await validateFindNew(ctx.request.query);
+
+    const { component, entityId, idsToOmit, page = 1, pageSize = 10, q } = ctx.request.query;
 
-    const modelDef = _component ? strapi.getModel(_component) : strapi.getModel(model);
+    const sourceModelUid = component || model;
```

Comment:
I thought we discussed this was not necessary as you can use a uid from the URL for components too as they are unique across compos & content types ?

Reply:
Yes, finally I came back to that solution because of permission reasons :/ So you are not able to see the component of a content type you don't have access to. What do you think?

Reply:
To check the permission indeed you would need the 2 🤔  but you aren't checking the permissions in the function from what I'm seeing?

Reply:
No, I realized that and I'm writing a commit for that :p

Reply:
done :)